ui/settingdialog.cpp

```diff
@@ -85,9 +85,6 @@ SettingDialog::SettingDialog(CameraManager *cameraManager, QWidget *parent)
 SettingDialog::~SettingDialog()
 {
     delete ui;
-    // Ensure all dynamically allocated memory is freed
-    qDeleteAll(settingTree->invisibleRootItem()->takeChildren());
-    delete this;
```

Comment:
`QTreeWidget`'s destructor already destroys all the items so no need to do it manually (also no `delete this` needed)